app/src/common/shared/org/mozilla/vrbrowser/browser/engine/SessionStore.java

```diff
@@ -149,8 +149,12 @@ public void onTrackingProtectionLevelUpdated(int level) {
         if (BuildConfig.DEBUG) {
             mStoreSubscription = ComponentsAdapter.get().getStore().observeManually(browserState -> {
                 Log.d(LOGTAG, "Session status BEGIN");
-                browserState.getTabs().forEach(tabSessionState -> Log.d(LOGTAG, "BrowserStore Session: " + tabSessionState.getId()));
-                mSessions.forEach(session -> Log.d(LOGTAG, "SessionStore Session: " + session.getId()));
```

Comment:
How was this causing the exception?

Reply:
`observeManually` get called in a pool thread every time the `BrowserStore` content changes so in some situations where we add/destroy sessions it might get called at the same time as a session is added/destroyed in the main thread. An easier way of reproducing it is just going to the tabs view and select and close all. 

We might also just remove that debugging code, I would like to move all that to a SessionStore unit test class, I'm doing some work on that but there is some refactoring involved and that's always tricky.